app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -1130,6 +1134,9 @@ private boolean checkForPendingAppHealthActions() {
         if (RecoveryMeasuresHelper.recoveryMeasuresPending()) {
             finishWithExecutionIntent();
             kickedOff = true;
+        } else if (UpdateActivity.isUpdateBlockedOnSync()) {
```

Comment:
this method (checkForPendingAppHealthActions) triggers any time we get back to the home screen, not just any time the app starts up for the first time, right? 

I think that means that failures to sync will keep forcing the user to keep trying to sync after , say, every form, even if they can't currently sync. I think this needs some way to identify that the user has attempted to sync already in the current login, and only force a sync if one hasn't already happened, so we can trigger the right behavior (sync before update), but only once per session (or once per update attempt. If the user restarts the update and it's blocked on sync it should reset that flag back to force a sync)